ArcGISRuntimeSDKQt_CppSamples/Geometry/CutGeometry/CutGeometry.cpp

```diff
@@ -0,0 +1,157 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "CutGeometry.h"
+
+#include "Map.h"
+#include "MapQuickView.h"
+#include "Graphic.h"
+#include "GraphicsOverlay.h"
+#include "GeometryEngine.h"
+#include "SpatialReference.h"
+#include "PolylineBuilder.h"
+#include "PolygonBuilder.h"
+#include "SimpleLineSymbol.h"
+#include "SimpleFillSymbol.h"
+
+using namespace Esri::ArcGISRuntime;
+
+CutGeometry::CutGeometry(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void CutGeometry::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<CutGeometry>("Esri.Samples", 1, 0, "CutGeometrySample");
+}
+
+void CutGeometry::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a map using the topographic basemap
+  m_map = new Map(Basemap::topographic(this), this);
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // Add a GraphicsOverlay
+  m_overlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(m_overlay);
+
+  // Create initial graphics
+  createGraphics();
+
+  // Set Viewpoint
+  m_mapView->setViewpointGeometry(m_lakeSuperiorGraphic->geometry());
+}
+
+// Cuts the Lake Superior polygon into two and symbolizes each output graphic
+void CutGeometry::cutPolygon()
+{
+  // perform the cut
+  QList<Geometry> geoms = GeometryEngine::cut(m_lakeSuperiorGraphic->geometry(), m_borderGraphic->geometry());
+
+  // create graphics for the U.S. side
+  Graphic* usSide = new Graphic(this);
+  usSide->setGeometry(geoms.at(0));
+  SimpleLineSymbol* outline = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor("blue"), 1.0f, this);
+  usSide->setSymbol(new SimpleFillSymbol(SimpleFillSymbolStyle::ForwardDiagonal, QColor("#40ff00"), outline, this));
+
+  // create graphics for the Canada side
+  Graphic* canadaSide = new Graphic(this);
+  canadaSide->setGeometry(geoms.at(1));
+  canadaSide->setSymbol(new SimpleFillSymbol(SimpleFillSymbolStyle::ForwardDiagonal, QColor("#ffff00"), outline, this));
+
+  // add graphics
+  m_overlay->graphics()->append(canadaSide);
+  m_overlay->graphics()->append(usSide);
+}
+
+// Create the two baseline graphics
+void CutGeometry::createGraphics()
+{
+  // Lake Superior Graphic
+  m_lakeSuperiorGraphic = new Graphic(this);
+  SimpleLineSymbol* outline = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor("blue"), 1.0f, this);
+  m_lakeSuperiorGraphic->setSymbol(new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(198, 204, 255, 178), outline, this));
+  m_lakeSuperiorGraphic->setGeometry(createLakeSuperiorPolygon());
```

Comment:
Same here, use the Graphic constructor instead for setting the symbol and graphic.